clients/java/signalr/src/main/java/HubConnection.java

```diff
@@ -23,19 +23,35 @@ public HubConnection(String url, Transport transport){
         this.protocol = new JsonHubProtocol();
         this.callback = (payload) -> {
 
-            InvocationMessage[] messages = protocol.parseMessages(payload);
+            HubMessage[] messages = protocol.parseMessages(payload);
 
             // message will be null if we receive any message other than an invocation.
             // Adding this to avoid getting error messages on pings for now.
-            for (InvocationMessage message : messages) {
-                if (message != null && handlers.containsKey(message.target)) {
-                    ArrayList<Object> args = gson.fromJson((JsonArray)message.arguments[0], (new ArrayList<Object>()).getClass());
-                    List<ActionBase> actions = handlers.get(message.target);
-                    if (actions != null) {
-                        for (ActionBase action: actions) {
-                            action.invoke(args.toArray());
+            for (HubMessage message : messages) {
+                switch (message.getMessageType()) {
+                    case INVOCATION:
+                        InvocationMessage invocationMessage = (InvocationMessage)message;
+                        if (message != null && handlers.containsKey(invocationMessage.target)) {
+                            ArrayList<Object> args = gson.fromJson((JsonArray)invocationMessage.arguments[0], (new ArrayList<Object>()).getClass());
+                            List<ActionBase> actions = handlers.get(invocationMessage.target);
+                            if (actions != null) {
+                                for (ActionBase action: actions) {
+                                    action.invoke(args.toArray());
+                                }
+                            }
                         }
-                    }
+                        break;
+                    case CLOSE:
+                    case STREAM_INVOCATION:
+                    case STREAM_ITEM:
+                    case CANCEL_INVOCATION:
+                    case COMPLETION:
+                    case ERROR:
```

Comment:
I'm not familiar with the `ERROR` message type

Reply:
Lol. Got a little trigger happy with message types and created a new one